libcircle.spec

```diff
@@ -0,0 +1,191 @@
+Name:    libcircle
+%global maj_ver 0.3
+Version: %{maj_ver}.0
+Release: 1%{?dist}
+
+Source: https://github.com/hpc/%{name}/releases/download/v%{maj_ver}/%{name}-%{version}.tar.gz
+URL: http://hpc.github.io/libcircle/
+Summary: A library used to distribute workloads
+License: BSD
+
+%global openmpi_ver openmpi3
+
+%if 0%{?fedora}
+%global openmpi_ver openmpi
+%endif
+
+BuildRequires:  check-devel
+BuildRequires:  doxygen
+BuildRequires:  graphviz
+
+%description
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+%package %{openmpi_ver}
+Summary:        Libcircle Open MPI libraries
+BuildRequires:  %{openmpi_ver}-devel
+
+%description %{openmpi_ver}
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+libcircle compiled with Open MPI
+
+%package mpich
+Summary:        Libcircle MPICH libraries
+BuildRequires:  mpich-devel environment-modules
+
+%description mpich
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+libcircle compiled with MPICH
+
+%package doc
+Summary:        Documentation for libcircle
+BuildArch:      noarch
+
+%description doc
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains documentation for libcircle
+
+%package %{openmpi_ver}-devel
+Summary:    Development headers and libraries for Open MPI libcircle
+Requires:   %{name}-%{openmpi_ver}%{?_isa} = %{version}-%{release}
+
+%description %{openmpi_ver}-devel
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains development headers and libraries for Open
+MPI ibcircle
+
+%package mpich-devel
+Summary:    Development headers and libraries for MPICH libcircle
+Requires:   %{name}-mpich%{?_isa} = %{version}-%{release}
+
+%description mpich-devel
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains development headers and libraries for
+MPICH libcircle
+
+%prep
+%setup -q
+
+%build
+mkdir %{openmpi_ver} mpich
+%global _configure ../configure
+
+pushd %{openmpi_ver}
+%if "%{openmpi_ver}" == "openmpi"
+%{_openmpi_load}
+%else
+%{_openmpi3_load}
+%endif
```

Comment:
It's tidier to define a macro at the top of the file using this conditional logic and then just using that macro here.